### PR TITLE
Spark 2.3 support -  update pom.xml for Spark 2.3

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
Add profile for Spark 2.3 in pom.xml

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -678,6 +678,65 @@
...
+              <sourceDirectory>src/main/spark2.3/antlr4</sourceDirectory>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>src/main/spark2.1</exclude>
+                <exclude>src/main/spark2.2</exclude>
```
Comment (JkSelf) [inline pom.xml]: @zhixingheyi-tian , After we create the directory "src/main/spark-2.3" for spark 2.3, we may also add "<exclude>src/main/spark2.3</exclude>" in the spark-2.1 and spark-2.2 profile of pom.xml to ensure not compile spark2.3 code when compile OAP with spark2.1 or spark2.2.

**Comment by carsonwang** [comment]
@zhixingheyi-tian , please address @JkSelf 's comment so we will go ahead and merge this.

**Comment by zhixingheyi-tian** [comment]
@gczsjdy 
Use the following command:

mvn clean -DskipTests -P spark-2.2 -Dcheckstyle.skip=true package
mvn clean -DskipTests -P spark-2.1 -Dcheckstyle.skip=true package

Test pass

**Comment by gczsjdy** [comment]
There should be `-Pspark-2.3` testing. Also, don't skip anything.

**Comment by zhixingheyi-tian** [comment]
About "-P spark-2.3". Currently skip the test code compilation. Because the test code has not adapted to spark 2.3 now.

The compile command is as follows:
Mvn clean -Dmaven.test.skip=true -P spark-2.3 -Dcheckstyle.skip=true packag

**Comment by carsonwang** [comment]
Thanks. Merging this first..

**Pull request merged**